main.tf

```diff
@@ -50,10 +50,21 @@ resource "google_compute_subnetwork" "subnetwork" {
   ip_cidr_range            = each.value.subnet_ip
   region                   = each.value.subnet_region
   private_ip_google_access = lookup(each.value, "subnet_private_access", "false")
-  enable_flow_logs         = lookup(each.value, "subnet_flow_logs", "false")
-  network                  = google_compute_network.network.name
-  project                  = var.project_id
-  description              = lookup(each.value, "description", null)
+  dynamic "log_config" {
+    for_each = lookup(each.value, "subnet_flow_logs", false) || lookup(each.value, "subnet_flow_logs_interval", null) != null || lookup(each.value, "subnet_flow_logs_sampling", null) != null || lookup(each.value, "subnet_flow_logs_metadata", null) != null ? [{
```

Comment:
Instead of this complex conditional, I think we should probably just key off `subnet_flow_logs` (requiring it to be `true` if you want flow logging enabled).

Reply:
So people would need to set `subnet_flow_logs` as well as the log config stuff?  
Or do you mean that `subnet_flow_logs` is no longer an option people can specify?  
Sorry for the confusion!

Reply:
Right, `subnet_flow_logs` would control activation while the other ones control its behavior.